File: recyclerview-swipe/src/main/java/com/yanzhenjie/recyclerview/swipe/touch/DefaultItemTouchHelper.java
/*
 * Copyright 2016 Yan Zhenjie
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.yanzhenjie.recyclerview.swipe.touch;

import android.support.v7.widget.helper.CompatItemTouchHelper;

/**
 * Created by Yolanda on 2016/4/19.
 */
public class DefaultItemTouchHelper extends CompatItemTouchHelper {

    private DefaultItemTouchHelperCallback mDefaultItemTouchHelperCallback;

    /**
     * Create default item touch helper.
     */
    public DefaultItemTouchHelper() {
        this(new DefaultItemTouchHelperCallback());
    }

    /**
     * @param callback the behavior of ItemTouchHelper.
     */
    private DefaultItemTouchHelper(DefaultItemTouchHelperCallback callback) {
        super(callback);
        mDefaultItemTouchHelperCallback = (DefaultItemTouchHelperCallback) getCallback();
    }

    /**
     * Set OnItemMoveListener.
     *
     * @param onItemMoveListener {@link OnItemMoveListener}.
     */
    public void setOnItemMoveListener(OnItemMoveListener onItemMoveListener) {
        mDefaultItemTouchHelperCallback.setOnItemMoveListener(onItemMoveListener);
    }

    /**
     * Get OnItemMoveListener.
     *
     * @return {@link OnItemMoveListener}.
     */
    public OnItemMoveListener getOnItemMoveListener() {
        return mDefaultItemTouchHelperCallback.getOnItemMoveListener();
    }

    /**
     * Set OnItemMovementListener.
     *
     * @param onItemMovementListener {@link OnItemMovementListener}.
     */
    public void setOnItemMovementListener(OnItemMovementListener onItemMovementListener) {
        mDefaultItemTouchHelperCallback.setOnItemMovementListener(onItemMovementListener);
    }

    /**
     * Get OnItemMovementListener.
     *
     * @return {@link OnItemMovementListener}.
     */
    public OnItemMovementListener getOnItemMovementListener() {
        return mDefaultItemTouchHelperCallback.getOnItemMovementListener();
    }

    /**
     * Set can long press drag.
     *
     * @param canDrag drag true, otherwise is can't.
     */
    public void setLongPressDragEnabled(boolean canDrag) {
        mDefaultItemTouchHelperCallback.setLongPressDragEnabled(canDrag);
    }

    /**
     * Get can long press drag.
     *
     * @return drag true, otherwise is can't.
     */
    public boolean isLongPressDragEnabled() {
        return mDefaultItemTouchHelperCallback.isLongPressDragEnabled();
    }


    /**
     * Set can long press swipe.
     *
     * @param canSwipe swipe true, otherwise is can't.
     */
    public void setItemViewSwipeEnabled(boolean canSwipe) {
        mDefaultItemTouchHelperCallback.setItemViewSwipeEnabled(canSwipe);
    }

    /**
     * Get can long press swipe.
     *
     * @return swipe true, otherwise is can't.
     */
    public boolean isItemViewSwipeEnabled() {
        return this.mDefaultItemTouchHelperCallback.isItemViewSwipeEnabled();
    }

    /**
     * Set OnItemStateChangedListener.
     *
     * @param onItemStateChangedListener {@link OnItemStateChangedListener}.
     */
    public void setOnItemStateChangedListener(OnItemStateChangedListener onItemStateChangedListener) {
        this.mDefaultItemTouchHelperCallback.setOnItemStateChangedListener(onItemStateChangedListener);
    }

    /**
     * Get OnItemStateChangedListener.
     *
     * @return {@link OnItemStateChangedListener}.
     */
    public OnItemStateChangedListener getOnItemStateChangedListener() {
        return this.mDefaultItemTouchHelperCallback.getOnItemStateChangedListener();
    }

}

File: recyclerview-swipe/src/main/java/com/yanzhenjie/recyclerview/swipe/touch/DefaultItemTouchHelper.kt
/*
 * Copyright 2016 Yan Zhenjie
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.yanzhenjie.recyclerview.swipe.touch

import android.support.v7.widget.helper.CompatItemTouchHelper

/**
 * Default item touch helper.
 *
 * Created by Yolanda on 2016/4/19.
 */
class DefaultItemTouchHelper : CompatItemTouchHelper {

  private val mDefaultItemTouchHelperCallback: DefaultItemTouchHelperCallback

  /**
   * Create default item touch helper.
   */
  constructor() : this(DefaultItemTouchHelperCallback())

  /**
   * Create default item touch helper.
   *
   * @param callback the behavior of ItemTouchHelper.
   */
  private constructor(callback: DefaultItemTouchHelperCallback): super(callback) {
    mDefaultItemTouchHelperCallback = callback
  }

  /**
   * Item move listener.
   */
  var onItemMoveListener: OnItemMoveListener?
    get() = mDefaultItemTouchHelperCallback.onItemMoveListener
    set(onItemMoveListener) {
      mDefaultItemTouchHelperCallback.onItemMoveListener = onItemMoveListener
    }

  /**
   * Item movement listener.
   */
  var onItemMovementListener: OnItemMovementListener?
    get() = mDefaultItemTouchHelperCallback.onItemMovementListener
    set(onItemMovementListener) {
      mDefaultItemTouchHelperCallback.onItemMovementListener = onItemMovementListener
    }

  /**
   * Long press drag enabled status.
   */
  var isLongPressDragEnabled: Boolean
    get() = mDefaultItemTouchHelperCallback.isLongPressDragEnabled
    set(canDrag) {
      mDefaultItemTouchHelperCallback.isLongPressDragEnabled = canDrag
    }

  /**
   * Item view swipe enabled status.
   */
  var isItemViewSwipeEnabled: Boolean
    get() = this.mDefaultItemTouchHelperCallback.isItemViewSwipeEnabled
    set(canSwipe) {
      mDefaultItemTouchHelperCallback.isItemViewSwipeEnabled = canSwipe
    }

  /**
   * Item state changed listener.
   */
  var onItemStateChangedListener: OnItemStateChangedListener?
    get() = this.mDefaultItemTouchHelperCallback.onItemStateChangedListener
    set(onItemStateChangedListener) {
      this.mDefaultItemTouchHelperCallback.onItemStateChangedListener = onItemStateChangedListener
    }

}
